### ISSUE TITLE
[BUG] Error when snapshot angular page

### ISSUE THREAD

**Issue by BrunoBeraudPW** [microsoft/playwright#20983]
Hello !

When I'm trying to take a snapshot of an angular page using Firefox it's generate an exception in my .net 6 application

You can reproduce that in trying to take snapshot of of the angular official website, it generate the same error than the one generated for my angular application

One thing weird is that it throw this error even with  the strict minimum to create HTML 5 page in an angular web app
```
<!DOCTYPE html>
<html>
   <head>
    </head>
    <body>
    </body>
</html>
```
And, if I remove the _body_ balise then there no more error...

On 1.27.2 version we don't have this issue (but we have a firefox bug that enforce us to use next version to get a more recent browser version...)

Only reproducible on Firefox (on chromium it's working)

Thanks !

**Context:**
- Playwright Version: 1.30.0
- Operating System: Windows 10
- Node.js version:  14.15.0
- Browser: Firefox
- net6.0 environment

**Code Snippet**
```
using var playwright = await Playwright.CreateAsync();
var browser = await playwright.Firefox.LaunchAsync();
var browserPage = await context.NewPageAsync();
await page.GotoAsync("https://angular.io/");
await page.ScreenshotAsync()
```

**error message**
`Unexpected token u in JSON at position 0`

**error stack trace**
```
at Microsoft.Playwright.Transport.Connection.InnerSendMessageToServerAsync[T](String guid, String method, Dictionary`2 dictionary) in /_/src/Playwright/Transport/Connection.cs:line 143
   at Microsoft.Playwright.Transport.Connection.WrapApiCallAsync[T](Func`1 action, Boolean isInternal) in /_/src/Playwright/Transport/Connection.cs:line 462
   at Microsoft.Playwright.Transport.Channels.PageChannel.ScreenshotAsync(String path, Nullable`1 fullPage, Clip clip, Nullable`1 omitBackground, Nullable`1 type, Nullable`1 quality, IEnumerable`1 mask, Nullable`1 animations, Nullable`1 caret, Nullable`1 scale, Nullable`1 timeout) in /_/src/Playwright/Transport/Channels/PageChannel.cs:line 395       
   at Microsoft.Playwright.Core.Page.ScreenshotAsync(PageScreenshotOptions options) in /_/src/Playwright/Core/Page.cs:line 592
   at PushToFactories.Application.Api.Services.SnapshotService.SnapshotService.TakeSnapshot(IPage page, String url) in /app/start/Services/SnapshotService/SnapshotService.cs:line 38
   at PushToFactories.Application.Api.Services.GenerateInnerPageJpegService.GenerateInnerPageJpegService.GeneratePage(Page page, Int32 resolutionInDpi, Guid projectId, IPage browserPage) in /app/start/Services/GenerateInnerPageJpegService/GenerateInnerPageJpegService.cs:line 30
   at PushToFactories.Application.Api.Services.PhotowebProductProductionService.<>c__DisplayClass15_0.<<ProcessDefaultJpg>b__0>d.MoveNext() in /app/start/Services/PhotowebProductProductionService.cs:line 207
   --- End of inner exception stack trace ---
```

**Comment by yury-s** [comment]
Can reproduce it with the following test in node.js:

```ts
import { test, expect } from "@playwright/test";

test('test', async({ page }) => {
    await page.goto("https://angular.io/");
    await page.screenshot({path: 's.png'});
})
```

Output:
```
Running 1 test using 1 worker
  1) [firefox] › example.spec.ts:3:5 › test ========================================================

    page.screenshot: Unexpected token u in JSON at position 0
    =========================== logs ===========================
    taking page screenshot
    ============================================================

      3 | test('test', async({ page }) => {
      4 |     await page.goto("https://angular.io/");
    > 5 |     await page.screenshot({path: 's.png'});
        |                ^
      6 | })


```

**Comment by rr13k** [comment]
+1 When can I repair it

**Comment by aslushnikov** [comment]
This is fixed in 1.31 - dupe of https://github.com/microsoft/playwright/issues/20434
